### PR TITLE
[Frontend] Init with Nuxtjs- A Station List index view

### LINKED ISSUE #88: Create station list with vue.js

### PR DESCRIPTION
**Pull request by kickapoo**
Fixes #88

Continued worked based on: https://github.com/openmeteo/enhydris/pull/87

### CODE REVIEW

**Comment by aptiko** [comment]
Note: Fields to show in the table are name, water basin and owner. (Later we may add "last update", but we don't have it yet; it's aptiko/openhi#23.)

**Comment by kickapoo** [comment]
https://github.com/openmeteo/enhydris/pull/91#issuecomment-452658647
screenshot 2019-01-10 17 59 50

File: frontend/components/dev/DevBox.vue [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+export default {
+  name: "DevBox",
+  data: function() {
+    return {
+      mode: utils.get_app_mode(),
+      apiRoot: utils.get_api_root()
+    };
+  }
+};
```
Comment (aptiko) [inline frontend/components/dev/DevBox.vue]: Use arrow function.

File: frontend/components/layout/Navbar.vue [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  name: "Navbar",
+  components: {
+    DevBox
+  },
+  data: function() {
+    return {
+      logoSrc: utils.get_app_logo(),
+      logoAlt: utils.get_app_logo_alt()
+    };
+  }
```
Comment (aptiko) [inline frontend/components/layout/Navbar.vue]: Use arrow function.

File: frontend/components/ui/SelectDivisionField.vue [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  props: {
+    value: [Number, String],
+  },
+  data() {
+    return {
+      loading: false,
+      bindVal: this.value,
+      divisions: []
+    }
+  },
```
Comment (aptiko) [inline frontend/components/ui/SelectDivisionField.vue]: Use arrow function, unless it's standard practice in vue to do it in another way. (Same thing in all components.)

File: frontend/pages/index.vue [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+          this.isPageLoading = false;
+          throw e;
+        });
+    },
+    makeMarkers: function(){
+      return maputils.stationsWithMarkers(this.curStations)
+    }
+  },
+  computed: {
+    curStations() {
```
Comment (aptiko) [inline frontend/pages/index.vue]: Don't use `let` here, use `var` (or use `let` here but also use it elsewhere, i.e. be consistent).

File: frontend/components/ui/SelectDivisionField.vue (hunk 3 above)
Comment (kickapoo) [reply frontend/components/ui/SelectDivisionField.vue]: https://stackoverflow.com/questions/42971081/use-arrow-function-in-vue-computed-does-not-work

File: frontend/components/layout/Navbar.vue (hunk 2 above)
Comment (kickapoo) [reply frontend/components/layout/Navbar.vue]: Navbar will be removed as mock up is not clear yet.

File: frontend/pages/index.vue (hunk 4 above)
Comment (kickapoo) [reply frontend/pages/index.vue]: `let` over `var`

File: frontend/utils/map.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
+const maputils = {
+  bbox_center: function(bbox){
+    // bounding_box center point
+    let box = bbox;
+    return [(box[1] + box[3]) / 2, (box[0] + box[2]) / 2];
+  },
+  stationsWithMarkers: function(data) {
```
[the remark this answers is not in the document]
Comment (kickapoo) [reply frontend/utils/map.js]: This a over simplified function to completed problem. Exists in this file in order to be updated

File: frontend/utils/map.js [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+        .slice(0, -1)
+        .split(" ")
+        .map(Number);
+      return [newpoint[1], newpoint[0]];
+    } else {
+      return [null, null];
+    }
+  }
+}
+export default maputils;
```
[the remark this answers is not in the document]
Comment (kickapoo) [reply frontend/utils/map.js]: I disagree. Use the same structure as the rest utils folder files

**Comment by kickapoo** [comment]
For reference  prettier: https://nuxtjs.org/guide/development-tools/#eslint-and-prettier

**Comment by aptiko** [comment]
There are still many things that have to be fixed, but we can merge this for a start.

Please squash all commits and use commit message `[Frontend] Create list of stations with vue/nuxt`.

**Comment by kickapoo** [comment]
Why this is still open even that is merged ?

**Comment by aptiko** [comment]
Apparently this wasn't automatically closed because of some change in the commits and then I failed to close it.

**Pull request closed without merging**